plugins/node/opentelemetry-instrumentation-ioredis/src/ioredis.ts

```diff
@@ -0,0 +1,93 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import type * as ioredisTypes from 'ioredis';
+import {
+  InstrumentationBase,
+  InstrumentationConfig,
+  InstrumentationNodeModuleDefinition,
+  isWrapped,
+} from '@opentelemetry/instrumentation';
+import { IORedisInstrumentationConfig } from './types';
+import { traceConnection, traceSendCommand } from './utils';
+import { VERSION } from './version';
+
+export class IORedisInstrumentation extends InstrumentationBase<
+  typeof ioredisTypes
+> {
+  static readonly DB_SYSTEM = 'redis';
+  readonly supportedVersions = ['>1 <5'];
+
+  constructor(protected _config: IORedisInstrumentationConfig = {}) {
+    super('@opentelemetry/instrumentation-ioredis', VERSION, _config);
+  }
+
+  public setConfig(
```

Comment:
What is this function for?

Reply:
I used the grpc instrumentation as example how to port and I found it also there: https://github.com/open-telemetry/opentelemetry-js/blob/b8afb044ff3e4bef8d9e375317f2501e82d1ff1e/packages/opentelemetry-instrumentation-grpc/src/grpc.ts#L68-L72

I'm fine with removing it.

Reply:
In general I'm not sure it's a good idea to allow changing the configuration after construction.

Reply:
Removed it. we can re-add it later if needed.

Reply:
I used this both in `http` and `grpc` instrumentation to be able to dinamically updated the config (only need a disable/enable to take effect) mostly for tests, not sure if it has real use cases in the wild though

Reply:
I see enable/disable quite risky as there is no gurantee that unwrap/wrap works as expected.

But if the config contains only stuff read once a API is called it should be ok to update without enable/disable. the more problematic part is then if a config changes happens in the middle of an transaction.

Reply:
I see this as opportunity to give user full control over the plugin. You would be able to enable plugin with different setting whenever you want without the need on restarting the service. We just have to replace shimmer - (just checked just one year passed since I created https://github.com/obecny/mpwrapper as alternative). This would be great especially in web environment or on mobile as you don't want to ask user to reload the page or close and open the app again. You would be able to change settings for the plugins fully remotely - this gives great possibilities for end user, A/B testing, etc. etc. - you can't easily replace version on user phone for example. But you can have it then the same on server too.

Reply:
> But if the config contains only stuff read once a API is called it should be ok to update without enable/disable. the more problematic part is then if a config changes happens in the middle of an transaction.

If the user update the config and have a big enough traffic (>100req/s for ex) that will happen for sure however i'm not sure to get why it's problematic ? The end user know that he updated the config so he can expect weird traces that happened in between both state

Reply:
> however i'm not sure to get why it's problematic

For simple instrumentations like ioredis this is for sure no issue. But there might be more complex ones where several low level operations build up one span and the same configuration has to be used across these operations to get a consistent result. In these cases it's needed to pass the current config along with the operation.

If we re-add `setConfig` we should clearly document and test what is needed to get it active.
If it goes via disable/enable also the risks (e.g. disable will not unwrap if someone else wrapped after us) should be clearly state. I would prefer to do this in a followup PR.

Reply:
I agree that if we want a `setConfig` feature it should be documented and properly added to the base class in a way that can be implemented consistently (possibly a `config_changed` event or similar). For now, I don't see any problem with removing it.

Reply:
I'm fine with removing it from this PR too since it wasn't in place, however do we want to refactor other instrumentations to remove it too ?